### ISSUE TITLE
Refactor Utils.extractVcs to not use FilePath

### ISSUE THREAD

**Issue by jetersen** [jfrog/jenkins-artifactory-plugin#213]
The pipeline steps should NOT need to run on an node just to extract git vcs info... 😭
The information is available from [git build data action](https://github.com/jenkinsci/git-plugin/blob/2e6d1bf0d10e04632f8153a3d2d9fc4d7d7c8bf2/src/main/java/hudson/plugins/git/util/BuildData.java#L65) or accessible from [build environment](https://github.com/jenkinsci/git-plugin/blob/2e6d1bf0d10e04632f8153a3d2d9fc4d7d7c8bf2/src/main/java/hudson/plugins/git/GitSCM.java#L1368-L1376).

https://github.com/RobiNino/jenkins-artifactory-plugin/blob/5accc853fdd6d786da913b510a5e6774d8449cff/src/main/java/org/jfrog/hudson/pipeline/common/Utils.java#L136-L142

Related to #212 as publishBuildInfo should not require a file path context ie. a node/agent.

The changes introduced in #185 is considered breaking changes previous pipeline steps now require to run on an node/agent.

**Comment by casz** [comment]
The changes introduced in #185 

break a pipeline like this where our shared library step `publishBuildInfoArtifactory` calls `rtServer.publishBuildInfo`

```
pipeline {
  agent none
  post {
    success {
      publishBuildInfoArtifactory()
    }
  }
  stages {
    stage('GitVersion') {
      agent {
        docker {
          image "artifactorydk.3shape.local/threeshapedocker/threeshape.msbuild:ltsc2019"
          args '-u ContainerAdministrator --entrypoint powershell'
          label "windows"
        }
      }
      steps {
        dotnetTool(tool: 'gitversion')
      }
    }
    stage('build and test') {
      parallel {
        stage('Windows DotNet Core SDK Nano') {
          agent {
            docker {
              image "mcr.microsoft.com/dotnet/core/sdk:3.0"
              label "windows"
            }
          }
          stages {
            stage('Nuget restore') {
              steps {
                powershell "dotnet restore"
              }
            }
            stage('Build solution') {
              steps {
                powershell "dotnet build -c:${configuration} --no-restore"
              }
            }

            stage('Pack') {
              steps {
                powershell "dotnet pack -p:PackageId=testing-3.0-core-nanoserver -c:${configuration} --no-build --no-restore --output '${nugetPackOutputDir}'"
              }
            }
            stage('Publish to Artifactory') {
              when {
                expression { publishToArtifactory == true }
              }
              steps {
                uploadNugetToArtifactory(nugetPackOutputDir: nugetPackOutputDir, nugetFeatures: 'nuget-features/testing-nugetcreator')
              }//steps
            }//stage
          }//stages
        }//windows stage
        stage('Linux') {
          agent {
            docker {
              image "mcr.microsoft.com/dotnet/core/sdk:3.0"
              label "linux"
            }
          }
          stages {
            stage('Nuget restore') {
              steps {
                sh "dotnet restore"
              }
            }
            stage('Build solution') {
              steps {
                sh "dotnet build -c:${configuration} --no-restore"
              }
            }

            stage('Pack') {
              steps {
                sh "dotnet pack -p:PackageId=testing-linux -c:${configuration} --no-build --no-restore --output '${nugetPackOutputDir}'"
              }
            }
            stage('Publish to Artifactory') {
              when {
                expression { publishToArtifactory == true }
              }
              steps {
                uploadNugetToArtifactory(nugetPackOutputDir: nugetPackOutputDir, nugetFeatures: 'nuget-features/testing-nugetcreator')
              }//steps
            }//stage
          }//stages
        }//linux stage
      }//parallel
    }//stage
  }//stages
}//pipeline
```

**Comment by eyalbe4** [comment]
This issue is fixed and released as part of version 3.13.1 of the Jenkins Artifactory Plugin. We would appreciate any feedback for the fix.

**Comment by jetersen** [comment]
@zionyx @MindaugasLaganeckas maybe your still interested in this? 🤔
